### PR TITLE
MGMT-12339: Add signature to monitor errors on cleanupInstallDevice

### PR DESCRIPTION
**Pull request by javipolo**
With the change to make disk cleanup non breaking, we need a way to monitor when those errors happen

This PR adds a new signature for checking that

Related PR:
https://github.com/openshift/assisted-installer/pull/583/files

/cc @omertuc

### CODE REVIEW

File: tools/add_triage_signature.py [hunk 1]
```diff
@@ -2037,6 +2037,44 @@ def _process_ticket(self, url, issue_key):
...
+        md = get_metadata_json(url)
+        cluster = md["cluster"]
+        cluster_id = cluster["id"]
+        triage_logs_tar = get_triage_logs_tar(triage_url=url, cluster_id=cluster_id)
+
+        for host in cluster["hosts"]:
+            host_id = host["id"]
+
+            try:
+                agent_logs = get_host_log_file(triage_logs_tar, host_id, "agent.logs")
```
Comment (omertuc) [inline tools/add_triage_signature.py:2068]: https://github.com/openshift/assisted-installer/pull/583 is modifying installer code, not agent code.

So need to look in `installer.logs` and not in `agent.logs`

Comment (javipolo) [reply tools/add_triage_signature.py]: Oops ... fixed!

File: tools/add_triage_signature.py [hunk 2]
```diff
@@ -2037,6 +2037,44 @@ def _process_ticket(self, url, issue_key):
...
+        for host in cluster["hosts"]:
+            host_id = host["id"]
+
+            try:
+                installer_logs = get_host_log_file(triage_logs_tar, host_id, "installer.logs")
+            except FileNotFoundError:
+                continue
+
+            if self.LOG_PATTERN.search(installer_logs):
+                self._update_triaging_ticket(
```
Comment (omertuc) [inline tools/add_triage_signature.py:2073]: this must be called only once per signature. So you need to collect info from all hosts and construct a table. See other signatures for an example

File: tools/add_triage_signature.py [hunk 3]
```diff
@@ -2037,6 +2037,44 @@ def _process_ticket(self, url, issue_key):
...
+            comment_identifying_string="h1. Error trying to perform cleanup on installation disk",
+        )
+
+    def _process_ticket(self, url, issue_key):
+        md = get_metadata_json(url)
+        cluster = md["cluster"]
+        cluster_id = cluster["id"]
+        triage_logs_tar = get_triage_logs_tar(triage_url=url, cluster_id=cluster_id)
+
+        for host in cluster["hosts"]:
```
Comment (omertuc) [inline tools/add_triage_signature.py:2064]:
```suggestion
        cluster = get_metadata_json(url)["cluster"]
        triage_logs_tar = get_triage_logs_tar(triage_url=url, cluster_id=cluster["id"])

        for host in cluster["hosts"]:
```

File: tools/add_triage_signature.py [hunk 4]
```diff
@@ -2037,6 +2037,44 @@ def _process_ticket(self, url, issue_key):
...
+            host_id = host["id"]
+
+            try:
+                installer_logs = get_host_log_file(triage_logs_tar, host_id, "installer.logs")
+            except FileNotFoundError:
+                continue
+
+            if self.LOG_PATTERN.search(installer_logs):
+                self._update_triaging_ticket(
+                    "cleanupInstallDevice function has failed. However, its failure does not block installation. Check logs to see if it is related to the installation failure"
```
Comment (omertuc) [inline tools/add_triage_signature.py:2074]: Can we include the actual line in the output for each host?

File: tools/add_triage_signature.py [hunk 5]
```diff
@@ -2037,6 +2037,44 @@ def _process_ticket(self, url, issue_key):
...
+    # https://github.com/openshift/assisted-installer/blob/51c021f3245ef1d9e1a50a3e31dc23350cdc5d32/src/installer/installer.go#L98
+    # Remember to maintain backwards compatibility if that format ever changes - create
+    # PATTERN_NEW and try to match both.
+    LOG_PATTERN = re.compile(r'msg="failed to prepare install device')
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(
+            *args,
+            **kwargs,
+            comment_identifying_string="h1. Error trying to perform cleanup on installation disk",
```
Comment (omertuc) [inline tools/add_triage_signature.py:2055]:
```suggestion
            comment_identifying_string="h1. Non-fatal error(s) occurred trying to perform best-effort cleanup on installation disk",
```

trying to make it sound less scary

File: tools/add_triage_signature.py (hunk 3 above)
Comment (javipolo) [reply tools/add_triage_signature.py]: Changed

File: tools/add_triage_signature.py (hunk 4 above)
Comment (javipolo) [reply tools/add_triage_signature.py]: Yep, done!

File: tools/add_triage_signature.py (hunk 5 above)
Comment (javipolo) [reply tools/add_triage_signature.py]: hahah, sure, changed!

File: tools/add_triage_signature.py [hunk 6]
```diff
@@ -2037,6 +2037,52 @@ def _process_ticket(self, url, issue_key):
...
+                installer_logs = get_host_log_file(triage_logs_tar, host_id, "installer.logs")
+            except FileNotFoundError:
+                continue
+
+            m = self.LOG_PATTERN.search(installer_logs)
+            if m:
+                hosts.append(
+                    OrderedDict(
+                        host=self._get_hostname(host),
+                        message=m.group("message"),
```
Comment (omertuc) [inline tools/add_triage_signature.py:2076]:
```suggestion
            if match := self.LOG_PATTERN.search(installer_logs):
                hosts.append(
                    OrderedDict(
                        host=self._get_hostname(host),
                        message=match.group("message"),
```

File: tools/add_triage_signature.py [hunk 7]
```diff
@@ -2037,6 +2037,52 @@ def _process_ticket(self, url, issue_key):
...
+class ErrorOnCleanupInstallDevice(Signature):
+    """
+    This signature monitors errors on cleanupInstallDevice function
+    """
+
+    # This has to be maintained to be the same format as
+    # https://github.com/openshift/assisted-installer/blob/51c021f3245ef1d9e1a50a3e31dc23350cdc5d32/src/installer/installer.go#L98
+    # Remember to maintain backwards compatibility if that format ever changes - create
+    # PATTERN_NEW and try to match both.
+    LOG_PATTERN = re.compile(r'time=".+" level=[a-z]+ msg="(?P<message>failed to prepare install device.*)"')
```
Comment (omertuc) [inline tools/add_triage_signature.py:2049]: Being overly specific with the searched pattern would just lead to more likely breakage on changes, IMO just finding the original substring is good enough

**Review by omertuc — CHANGES_REQUESTED** [review]

File: tools/add_triage_signature.py (hunk 6 above)
Comment (javipolo) [reply tools/add_triage_signature.py:2076]: didnt, know about := operator in python, sweet!

Changed

File: tools/add_triage_signature.py (hunk 7 above)
Comment (javipolo) [reply tools/add_triage_signature.py:2049]: Sure, changed!

**Comment by javipolo** [comment]
> /hold until other pr merges

which other PR?
https://github.com/openshift/assisted-installer/pull/597 ?

**Pull request merged**